skbio/alignment/_alignment.py

```diff
@@ -1662,6 +1665,189 @@ def to_phylip(self, map_labels=False, label_prefix=""):
 
         return '\n'.join(result), new_id_to_old_id
 
+    def heatmap(self, value_map,
+                legend_labels=('Minimum', 'Median', 'Maximum'), fig_size=None,
+                cmap=None, sequence_order=None):
+        """Plot alignment as a heatmap.
+
+        Plot the alignment as a heatmap, coloring heatmap cells using the
+        character mapping in `value_map`. The x-axis is labeled by majority
+        consensus, and the y-axis is labeled by sequence IDs (first, last, and
+        every third sequence ID are displayed).
+
+        Parameters
+        ----------
+        value_map : dict, collections.defaultdict
+            Dictionary mapping characters in the alignment to numeric values.
+            ``KeyErrors`` are not caught, so all possible values should be in
+            `value_map`, or it should be a ``collections.defaultdict`` which
+            can, for example, default to ``nan``.
+        legend_labels : iterable, optional
```

Comment:
@gregcaporaso does the behavior described for `legend_labels` make sense?

In your original cookbook implementation, all of the values in `value_map` were used to compute the minimum, median, and maximum. This could produce strange behavior if a value in `value_map` wasn't used in the heatmap. For example, if `value_map` had an unused mapping to a maximum value, the "maximum" label wouldn't show up in the legend because a different (smaller) maximum value was used in the heatmap/colorbar.

Also, if a `defaultdict` was supplied, mappings using the default value weren't being considered when computing min/median/max. For example, suppose the `defaultdict` had a default value that happened to be the true maximum, the "maximum" label would be placed in the wrong spot.